shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/core/metadata/FederateTableMetadata.java

```diff
@@ -88,4 +91,13 @@ private DataSource getActualDataSource(final Map<String, DataSource> dataSources
         }
         return dataSources.get(result);
     }
+    
+    /**
+     * Get rel data type field.
+     * 
+     * @return rel data type field collection
+     */
+    public List<RelDataTypeField> getRelDataTypeField() {
```

Comment:
I have no idea how long a bunch of calls for `relProtoDataType.apply(TYPE_FACTORY)` will take up. But suppose we just wanna get the list of columns, do you think it will be better to cache these column names here from `TableMetaData` or this function?

Reply:
> I have no idea how long a bunch of calls for `relProtoDataType.apply(TYPE_FACTORY)` will take up. But suppose we just wanna get the list of columns, do you think it will be better to cache these column names here from `TableMetaData` or this function?

@tristaZero Yes, this does have performance problems, and I will consider caching column names.